src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3966,6 +4027,41 @@ export class StudyViewPageStore
         this.changeChartVisibility(uniqueKey, false);
     }
 
+    public getGenePanelDataCount(
+        chartMeta: ChartMeta
+    ): MobxPromise<ClinicalDataCountSummary[]> {
+        const suffix = chartMetaToProfile(chartMeta);
+
+        return remoteData<ClinicalDataCountSummary[]>({
+            await: () => [this.genePanelCounts],
+            invoke: async () => {
+                const unfCounts = this.genePanelCounts.result!;
+                const countsForProfile = unfCounts
+                    .filter(item => {
+                        return item.molecularProfileId.endsWith(suffix);
+                    })
+                    .map(i => {
+                        return {
+                            attributeId: i.molecularProfileId,
+                            counts: i.counts,
+                        };
+                    });
+
+                let data = countsForProfile[0];
+                let counts: ClinicalDataCount[] = [];
+                let attributeId: string = '';
+                if (data !== undefined) {
+                    counts = data.counts;
+                    attributeId = data.attributeId;
+                    if (!this.chartToUsedColors.has(attributeId))
+                        this.chartToUsedColors.set(attributeId, new Set());
```

Comment:
we shouldn't update state from inside of invoke method like this.  generally invoke only returns a value, which is used by refererer.